### PR TITLE
azurerm_api_management - support key_vault_id's without a version

### LINKED ISSUE #4408: API Management custom domain key vault address incorrectly validates
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
1.30.0

### Affected Resource(s)

* `azurerm_api_management`

### Terraform Configuration Files

```hcl
data "azurerm_key_vault_secret" "certs" {
  name         = "ssl-cert"
  key_vault_id = data.azurerm_key_vault.certs.id
}

locals {
  cert_url = replace(data.azurerm_key_vault_secret.certs.id,"/${data.azurerm_key_vault_secret.cert.version}","")
}

resource "azurerm_api_management" "api" {
...
  hostname_configuration {
    proxy {
      host_name    = "api.hostname.com"
      key_vault_id = local.cert_url
    }

    portal {
      host_name    = "developers.hostname.com"
      key_vault_id = local.cert_url
    }
  }
...
```

### Debug Output
Error: Error parsing Key Vault Child ID: Azure KeyVault Child Id should have 3 segments, got 2: 'secrets/ssl-cert'

### Expected Behavior
The key vault url should not require the version. By requiring a version, it defeats the purpose of using key vault, which allows new certs to be uploaded without requiring a new deployment of api management

It's also not possible to obtain a versionless url from the secret datasource without using the above function.

### Actual Behavior

Using the Azure portal sets the versionless url, which Terraform will then undo.

The provider uses the Keyvault validation function for a secret that requires 3 parts. In this instance, on 2 parts are required & are preferred. This validation was appears to have been added in #2189

### Steps to Reproduce

1. Define an api management resource with a custom domain.
1. Upload a certificate for the custom domain to a key vault store
1. Link the key vault cert to the custom domain

### References

* #0000

### PR DESCRIPTION
**Pull request by sirlatrom**
Fixes #4408.

### CODE REVIEW

**Comment by sirlatrom** [comment]
@katbyte As with previous PRs, I've not been able to run acceptance tests but have verified functionality within my org's tenant.

File: azurerm/internal/services/apimanagement/resource_arm_api_management.go [hunk 1]
```diff
@@ -1058,7 +1058,7 @@ func apiManagementResourceHostnameSchema(schemaName string) map[string]*schema.S
 		"key_vault_id": {
 			Type:         schema.TypeString,
 			Optional:     true,
-			ValidateFunc: azure.ValidateKeyVaultChildId,
+			ValidateFunc: azure.ValidateKeyVaultChildIdVersionOptional,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/apimanagement/resource_arm_api_management.go:1061]: Out of interest - when provisioning this, does the Azure API return the versionless-URI - or the latest version being used?

Comment (sirlatrom) [reply azurerm/internal/services/apimanagement/resource_arm_api_management.go:1061]: It returns the versionless URI, fortunately.

**Comment by sirlatrom** [comment]
> Hey @sirlatrom 
> 
> Thanks for this PR - taking a look through this mostly LGTM, however can we add an acceptance test covering provisioning with a versionless secret to confirm that this works?
> 
> Thanks!

I'd like to, but there's a catch 22: Currently, only System Assigned Identity is used against the Key Vault, and you _can_ of course use an `azurerm_key_vault_access_policy` to grant the API Management service's System Assigned Identity access to the Key Vault.

But if you create the API Management service with the Key Vault references, versioned or not, they are evaluated _during_ creation of the API Management service, preventing us from creating the access policy ahead of time.

I've tried with User Assigned Identity and confirmed it is not used for Key Vault access. I've got the impression that User Assigned Identity is currently only used to access API backends.

Is there a way that I can have two phases in the acceptance test? In our pipeline we're currently working around the issue by using a variable to conditionally set the `key_vault_id` field the first time, conditional on the existence of the relevant `azurerm_key_vault_access_policy` resource.

**Comment by sirlatrom** [comment]
@tombuildsstuff In 31e4993, I've added acceptance tests setting the `key_vault_uri` to either a versioned or a versionless URI.

**Comment by sirlatrom** [comment]
@tombuildsstuff @katbyte Rebased this on master after other tests were merged, should apply cleanly now.

**Pull request merged**